client/src/details/SFSUrlComponents.cpp

```diff
@@ -3,57 +3,102 @@
 
 #include "SFSUrlComponents.h"
 
+#include "ErrorHandling.h"
+#include "ReportingHandler.h"
+
+#include <curl/curl.h>
+
 using namespace SFS::details;
 
 namespace
 {
+struct CurlEscapedString
+{
+  public:
+    CurlEscapedString(const std::string& str, const ReportingHandler& handler)
+    {
+        m_escapedString = curl_easy_escape(nullptr /*ignored*/, str.c_str(), 0 /*use strlen*/);

```

Comment:
std::string has an embedded length, so it's inefficient to use strlen.

cf. https://github.com/libcpr/cpr/blob/1ad1cd2daf138aefdcdb1c35d99608387d230ec1/cpr/curlholder.cpp#L33